### DISCUSSION TITLE
V3 - WireableSynth Exception on setting data on Wireable class in test

### DISCUSSION THREAD

**Discussion by Naoray** [livewire/livewire#5989] [7. Bugs (archived)] [answered]
The `->fill()` call in the following test causes an `Exception: This synth doesn't support setting properties: Livewire\Features\SupportWireables\WireableSynth` error. It used to work fine in Livewire v2. I suppose it has something to do with setting a property on a wireable object directly instead of setting the whole object at once. Therefore I’ve also tried setting the whole data object with `->set(‘customer.credit_payment_info’, new PaymentInfoData(..))`, which led to another Error `Error: Cannot use object of type PaymentInfoData as array`.

```php
/** @test */
public function refresh_component_is_emitted_when_data_was_submitted_successfully()
{
    Livewire::actingAs($this->user)
        ->test(PaymentInfo::class, [
            'booking' => $this->user->booking,
        ])
        ->fill([
            'customer.credit_payment_info.method' => PaymentInfoData::PAYPAL,
            'customer.credit_payment_info.account_owner' => 'Test owner',
            'customer.credit_payment_info.paypal_email' => 'some@mail.com',
        ])
        ->call('submit')
        ->assertDispatched('refreshComponent');
}

class Customer extends Model
{
    protected $casts = [
        'credit_payment_info' => PaymentInfoData::class,
   ];
}

// using spatie/laravel-data (https://github.com/spatie/laravel-data)
class PaymentInfoData extends Data implements Wireable
{
    public function toLivewire(): array
    {
        return $this->toArray();
    }

    public static function fromLivewire($value): static
    {
        return app(DataFromSomethingResolver::class)
            ->ignoreMagicalMethods('fromLivewire')
            ->execute(static::class, $value);
    }

    //...
}
```

Can someone point me in the right direction on how to solve this issue?

Thank you in advance 🙏

**Comment by joshhanley** [comment]
@Naoray thanks for reporting! Any chance you could submit a failing test PR so we can investigate? Thanks!

**Reply by levu42** [reply]
@joshhanley I just created a PR with failing tests https://github.com/livewire/livewire/pull/6559 - this is something that happens now for me on a project I'm updating from v2 to v3, seems like a regression, updating wireables seems to be broken in v3 now.

**Reply by Naoray** [reply]
thanks @levu42 for creating the PR!

**Comment by ejonasson** [comment, marked as answer]
@Naoray I encountered the same issue with the `spatie/laravel-data` package. I ended up adding my own [synthesizer](https://livewire.laravel.com/docs/synthesizers#understanding-synthesizers) for my laravel-data Wireable properties.

I ended up writing the following: https://gist.github.com/ejonasson/0e437179fa20e5311073e7ef266e8b02 (I have not used this extensively yet so can't guarantee it's 100% accurate). 

So I'm not clear if this is something that needs an update on the Spatie side, or in Livewire v3.

**Reply by dennisvandalen** [reply]
Thanks! This did the trick. Don't forget to [register the synthesizer](https://livewire.laravel.com/docs/synthesizers#registering-a-custom-synthesizer). 

```
class AppServiceProvider extends ServiceProvider
{
    /**
     * Bootstrap any application services.
     */
    public function boot(): void
    {
        Livewire::propertySynthesizer(DataObjectSynth::class);
    }
}
```

**Reply by Naoray** [reply]
Thank you, @ejonasson. I'm marking this as the solution for now, even though it's a temporary fix. It should ideally work seamlessly by default due to the utilization of the `WireableData` trait provided by the `spatie/laravel-data` package.

**Reply by riomigal** [reply]
@ejonasson 

Thank you! Saved me a lot of time and I didn't encounter any issues until now.
